gilded-rose-kata/src/main/kotlin/models/Item.kt

```diff
@@ -0,0 +1,25 @@
+package models
+
+import enums.*
+import interfaces.*
+
+abstract class Item(val name: String, var sellIn: Int, quality: Int): ItemQualityModifier {
```

Comment:
Inheritance combined with mutable state can easily become a maintanance burden, as a good principle I would avoid it if possible - can we think in a alternative design for this hierarchy?

Reply:
Nothing really comes to my mind as a response to that (TBH)
I will play with the code and see if it rings any bells.

I will obviously give a close look to what you said. 

Just as a clarification. In your words: 

> Inheritance combined with mutable state can easily become a maintanance burden, as a good principle I would avoid it if possible - can we think in a alternative design for this hierarchy?

**Inheritence:** _ItemQualityModifier_ interface.
**mutable state:** _ItemQualityModifier_ quality variable
**this hierarchy:** The abstract class _Item_?